casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -792,4 +792,56 @@ object BlockAPI {
     )
   }
 
+  def getFromRuntime[F[_]: Concurrent: EngineCell: BlockStore, A, O](
+      blockHash: String,
+      source: A,
+      usePreStateHash: Boolean,
+      getFunc: (RuntimeManager[F], A, ByteString) => F[O]
+  ): F[O] =
+    for {
+      block <- BlockStore[F].getUnsafe(blockHash.unsafeHexToByteString)
+      stateHash = if (usePreStateHash) block.body.state.preStateHash
+      else block.body.state.postStateHash
+      engine <- EngineCell[F].read
+      getData = (casper: MultiParentCasper[F]) =>
+        for {
+          runtimeManager <- casper.getRuntimeManager
+          output         <- getFunc(runtimeManager, source, stateHash)
+        } yield output
+      result <- engine.withCasper(
+                 getData,
+                 Concurrent[F].raiseError[O](new Exception("Casper instance is not ready."))
+               )
+    } yield result
+
+  def getDataAtPar[F[_]: Concurrent: EngineCell: BlockStore](
```

Comment:
Let's keep it simple and just expose access to data, continuations are not needed right now.

This function can accept `RuntimeManager` directly (available in Setup) which can spawn new Runtimes. It's not necessary to have EngineCell and Casper instance.